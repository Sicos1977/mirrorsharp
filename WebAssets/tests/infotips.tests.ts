import { TestDriver, timers } from './test-driver';

const waitForHover = async (driver: TestDriver) => {
    timers.advanceTimersByTime(1000); // actual is 750, but just in case
    await driver.completeBackgroundWork();
};

test('hover requests infotip', async () => {
    const driver = await TestDriver.new({ text: 'a b c' });
    const cmView = driver.getCodeMirrorView();
    cmView.posAtCoords = () => 0;
    cmView.coordsAtPos = () => ({ left: 0, right: 0, top: 0, bottom: 0 });

    driver.domEvents.mousemove(cmView.contentDOM.firstChild!);
    await waitForHover(driver);

    expect(driver.socket.sent).toContain('I0');
});