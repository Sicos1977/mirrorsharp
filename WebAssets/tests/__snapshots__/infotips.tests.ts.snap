// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`hover applies received infotip 1`] = `"<div class=\\"mirrorsharp-parts-section\\"><span class=\\"mirrorsharp-infotip-icon mirrorsharp-infotip-icon-delegate mirrorsharp-infotip-icon-public\\"></span><span class=\\"cmt-keyword\\">delegate</span><span class=\\"cmt-space\\"> </span><span class=\\"cmt-keyword\\">void</span><span class=\\"cmt-space\\"> </span><span class=\\"cmt-namespace\\">System</span><span class=\\"cmt-punctuation\\">.</span><span class=\\"cmt-delegate\\">EventHandler</span><span class=\\"cmt-punctuation\\">(</span><span class=\\"cmt-keyword\\">object</span><span class=\\"cmt-space\\"> </span><span class=\\"cmt-parameter\\">sender</span><span class=\\"cmt-punctuation\\">,</span><span class=\\"cmt-space\\"> </span><span class=\\"cmt-namespace\\">System</span><span class=\\"cmt-punctuation\\">.</span><span class=\\"cmt-class\\">EventArgs</span><span class=\\"cmt-space\\"> </span><span class=\\"cmt-parameter\\">e</span><span class=\\"cmt-punctuation\\">)</span></div><div class=\\"mirrorsharp-parts-section\\"><span class=\\"cmt-text\\">Represents the method that will handle an event that has no event data.</span></div>"`;
